include/deal.II/grid/grid_tools.h

```diff
@@ -323,12 +330,21 @@ namespace GridTools
    * Should this function be provided, sensible results can only be expected
    * if all coefficients are positive.
    *
+   * @param[in] solve_for_position_field If set to <code>true</code>, the
+   * minimization problem is formulated with respect to the final vertex positions
+   * as opposed to their displacement. Note that this case may lead to non-zero
+   * displacement of un-constrained vertices even when the @p new_points
+   * don't specify any deformation. That is because the minimum corresponds to
+   * smooth vertex position field, instead if their displacement. Thereby this
+   * case can be considered as a variation of Laplacian smoothing technique.
```

Comment:
I don't think I believe this is correct, for the reason we're discussing in the general discussion section of this PR.